src/consumption.cpp

```diff
@@ -1095,6 +976,134 @@ void Character::modify_morale( item &food, const int nutr )
         add_morale( MORALE_FOOD_GOOD, fun.first, fun.second, morale_time, morale_time / 2, false,
                     food.type );
     }
+
+    bool spoiled = food.rotten();
+
+    // Morale bonus for eating unspoiled food at a nearby table
+    if( !spoiled && !food.has_flag( flag_ALLERGEN_JUNK ) ) {
+        bool has_table_nearby = false;
+        bool has_chair_nearby = false;
+        for( const tripoint &pt : g->m.points_in_radius( pos(), 1 ) ) {
+            if( g->m.has_flag_furn( flag_FLAT_SURF, pt ) || g->m.has_flag( flag_FLAT_SURF, pt ) ||
+                ( g->m.veh_at( pt ) && ( g->m.veh_at( pt )->vehicle().has_part( "KITCHEN" ) ||
+                                         g->m.veh_at( pt )->vehicle().has_part( "FLAT_SURF" ) ) ) ) {
+                has_table_nearby = true;
+            }
+            if( g->m.has_flag_furn( flag_CAN_SIT, pt ) || g->m.has_flag( flag_CAN_SIT, pt ) ||
+                ( g->m.veh_at( pt ) && ( g->m.veh_at( pt )->vehicle().has_part( "SEAT" ) ) ) ) {
+                has_chair_nearby = true;
+            }
+        }
+        if( has_chair_nearby && has_table_nearby ) {
+            if( has_trait( trait_TABLEMANNERS ) ) {
+                rem_morale( MORALE_ATE_WITHOUT_TABLE );
+                add_morale( MORALE_ATE_WITH_TABLE, 3, 3, 3_hours, 2_hours, true );
+            } else {
+                add_morale( MORALE_ATE_WITH_TABLE, 1, 1, 3_hours, 2_hours, true );
```

Comment:
This branching can be simplified. Just note that those function invocations differ only by the value of the bonus.

Reply:
I didn't do anything with this one; seems pretty minor and it's quite clear as-is.